azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -1111,16 +1112,20 @@ private void configureJobLevelMetrics(final JobRunner jobRunner) {
     jobRunner.addListener(JmxJobMBeanManager.getInstance());
   }
 
-  public void pause(final String user) {
+  public void pause(final String user) throws IllegalStateException {
     synchronized (this.mainSyncObj) {
-      if (!this.flowFinished) {
-        this.logger.info("Flow paused by " + user);
+      this.logger.info("Flow paused by " + user);
```

Comment:
Nit: would you like to say "flow pause requested by user"? It's not given that it will be paused at this point, right?

Reply:
I'll change the message, thx.